specs/core/0_fork-choice.md

```diff
@@ -118,9 +133,13 @@ def get_latest_attesting_balance(store: Store, root: Hash) -> Gwei:
 ```python
 def get_head(store: Store) -> Hash:
     # Execute the LMD-GHOST fork choice
-    head = store.justified_root
+    head = store.justified_checkpoint.root
+    justified_slot = get_epoch_start_slot(store.justified_checkpoint.epoch)
     while True:
-        children = [root for root in store.blocks.keys() if store.blocks[root].parent_root == head]
+        children = [
+            root for root in store.blocks.keys()
+            if store.blocks[root].parent_root == head and store.blocks[root].slot > justified_slot
```

Comment:
Just checking `parent_root` should be enough. But if we are doing both, maybe swap the slot check to be first (much faster)?

Reply:
It isn't actually enough. There is a case where the block is in the block tree since latest justified block but is from an epoch prior to latest justified epoch.

Example:
* Block A at slot 60
* Block B (parent A) at slot 61
* Block A is justified for epoch 1 (slot 64)
* The block tree under consideration is now descendants of Block A but exclusively after the epoch boundary slot of epoch 1.
* Thus Block B should not be in consideration.

That said, we can flip the two bool checks for speed